### PR TITLE
init Feature gm 

### PR DESCRIPTION
**Pull request by HaoXuan40404**
1. Modify CMakeList
*  Replaces OpenSSL with tassl.cmake, it replaces all files under libdevcrypto with libinitializer / SecureInitializer.cpp and replaces them with the files required by the national secret.
2. Libdevcrypto
*  Adds SM2, SM3 and SM4 packet encryption method
*  sha3.h and hash.h are merged and are currently in hash.h.
*  Common.h defines the parameters of the national and non-national signatures, and moves the signature constructor implementation to common.cpp.
*  Removed useless interfaces in AES.h, cryptopp.h, hash.h, common.h. Deleted useless rsa files
Libethcore module
*  The process of constructing a signature from RLP in a transaction is adjusted.
*  Adjusted the process of ecrecover in Precompiled.cpp and put it in common.h.

3.Libinitializer module
*  Added the file SecureInitializerGm.cpp when the national secret node SSL was established.
Libstorage module modification

4.Libstorage
*  Optimize magic number to map

### CODE REVIEW

File: tools/build_chain.sh [hunk 1]
```diff
@@ -239,10 +272,11 @@ gen_node_cert() {
     dir_must_exists "$agpath"
     file_must_exists "$agpath/agency.key"
     check_name agency "$agency"
-    dir_must_not_exists "$ndpath"
-    check_name node "$node"
 
     mkdir -p $ndpath
+    dir_must_exists "$ndpath"
+    check_name node "$node"
+
```
Comment (bxq2011hust) [inline tools/build_chain.sh]: The funtion is offered by others, can we just let it is?

File: cmake/ProjectBoost.cmake [hunk 2]
```diff
@@ -44,7 +44,7 @@ set(BOOST_BUILD_FILES ${BOOST_LIB_PREFIX}chrono.a ${BOOST_LIB_PREFIX}date_time.a
         ${BOOST_LIB_PREFIX}thread.a ${BOOST_LIB_PREFIX}program_options.a)
 set(BOOST_CXXFLAGS "cxxflags=-Wa,-march=generic64")
 
-set(CMAKE_ARGS -DOPENSSL_INCLUDE_DIRS=${OPENSSL_INCLUDE_DIRS})
+set(BOOST_CXXFLAGS "cxxflags=-Wa,-march=generic64")
```
Comment (bxq2011hust) [inline cmake/ProjectBoost.cmake]: Is this duplicated as before?

Comment (HaoXuan40404) [reply cmake/ProjectBoost.cmake]: delete it now

File: test/unittests/libethcore/ABI.cpp [hunk 3]
```diff
@@ -38,6 +39,14 @@ namespace test
 {
 BOOST_FIXTURE_TEST_SUITE(ABITest, TestOutputHelperFixture)
 
+void Test_abi_gm(const string& _str)
```
Comment (zhifubao) [inline test/unittests/libethcore/ABI.cpp]: pls delete it

Comment (HaoXuan40404) [reply test/unittests/libethcore/ABI.cpp]: done

**Pull request closed without merging**